lazybot/commands/__init__.py

```diff
@@ -7,8 +7,88 @@
 for a discord bot.
 """
 
-from .CommandColor import CommandColor
-from .CommandGradYear import CommandGradYear
-from .CommandHelp import StraightforwardHelp
-from .MiscFun import MiscFun
-from .UpdateChecker import UpdateChecker
+import inspect
+import os
+import sys
+import types
+
+import importlib.util
+from importlib.machinery import SourceFileLoader
+
+from discord.ext.commands import Cog
+
+_COMMANDS = []
+
+
+def all_commands():
+    """
+    Provide a copy of the list of commands.
+    """
+    return list(_COMMANDS)
+
+
+def load_commands():
+    """
+    Loads the commands that are contained within this package.
+    """
+
+    directory = os.path.dirname(os.path.realpath(__file__))
+
+    modules = list(
+        {
+            command_file
+            for command_file in os.listdir(directory)
+            if command_file.endswith(".py") and command_file != "__init__.py"
+        }
+    )
+
+    for module in modules:
+        path = os.path.join(directory, module)
+        modname = os.path.splitext(os.path.basename(path))[0]
+
+        try:
+            loader = SourceFileLoader(f"{__name__}.{modname}", path)
+            command_module = type.ModuleType(loader.name)
+            loader.exec_module(command_module)
+        except (ImportError, SyntaxError) as import_error:
+            print(
+                f"Unable to load command from {module}: {import_error}", file=sys.stderr
+            )
+        else:
+            # Load all the contents of the modules and check if they are
+            # subclasses of Cog. IF they are, add them to the commands list.
+            module_contents = [
+                getattr(command_module, content_name)
+                for content_name in dir(command_module)
+            ]
+            _COMMANDS.append(
+                {
+                    command_class
+                    for command_class in module_contents
+                    if _is_loaded_subclass(command_class, Cog)
+                }
+            )
```

Comment:
`_COMMANDS.extend(list(
                {
                    command_class
                    for command_class in module_contents
                    if _is_loaded_subclass(command_class, Cog)
                }
            ))`